### PR TITLE
feat: [CDS-85789]: Enabling Template feature for free and community accounts

### PR DESCRIPTION
**Pull request by meenaravichandran1**
## Describe your changes
Enabling Template feature for free accounts

## Test cases
1. Create new remote pipeline with a template 
2. Create new remote pipeline with a template with runtime input and expressions
3. Ability to see existing templates
4. Force deletion
5. Delete a template
6. Import template from git
7. Change variable in the template and check if it's reflected in the pipeline 
8. Create a step template, refer a pipeline, run pipeline
9. Create a stage template, refer a pipeline, run pipeline
10. Create new inline pipeline with a template
11. Create new inline pipeline with a template with runtime input and expressions
12. Create a step group template, refer a pipeline, run pipeline
13. Create a deployment template, refer a pipeline, run pipeline
14. Reference template within template
15. Change current template 
16. Check references for template

Validated in local and all the above test cases PASSED

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56238)

### CODE REVIEW

**Comment by sahilhindwani-harness** [comment]
*[`925-enforcement-service/src/main/resources/features/pipeline-features.yml` line 66 at r1](https://reviewable.io/reviews/harness/harness-core/56238#-NknhuYq-lDE4dvGE42X:-NknhuYq-lDE4dvGE42Y:b-jco0u6) ([raw file](https://github.com/harness/harness-core/blob/dd7a09294f13893a01f7ed2042ebe2d44b464b27/925-enforcement-service/src/main/resources/features/pipeline-features.yml#L66)):*
> ```YAML
>       COMMUNITY:
>         restrictionType: AVAILABILITY
>         enabled: true
> ```

Do we ship template service in community?

**Comment by meenaravichandran1** [comment]
> _[`925-enforcement-service/src/main/resources/features/pipeline-features.yml` line 66 at r1](https://reviewable.io/reviews/harness/harness-core/56238#-NknhuYq-lDE4dvGE42X:-NknhuYq-lDE4dvGE42Y:b-jco0u6) ([raw file](https://github.com/harness/harness-core/blob/dd7a09294f13893a01f7ed2042ebe2d44b464b27/925-enforcement-service/src/main/resources/features/pipeline-features.yml#L66)):_
> 
> > ```yaml
> >       COMMUNITY:
> >         restrictionType: AVAILABILITY
> >         enabled: true
> > ```
> 
> Do we ship template service in community?

yes @sahilhindwani-harness. We are enabling it for free and community accounts. Is there any dependency we should be aware of?

**Comment by rohitk-h** [comment]
- Can you please add some information on the scenarios tested ? 
- Do we have any documentation ticket/link about this change?

**Comment by meenaravichandran1** [comment]
> * Can you please add some information on the scenarios tested?
> * Do we have any documentation ticket/link about this change?

1. I am not aware if we can test this in some test accounts before enabling it. @sahilhindwani-harness can you confirm?
2. The linked ticket to this PR, https://harness.atlassian.net/browse/CDS-85789 has the details

File: 925-enforcement-service/src/main/resources/features/pipeline-features.yml [hunk 1]
```diff
@@ -57,13 +57,13 @@ features:
...
         restrictionType: AVAILABILITY
-        enabled: false
+        enabled: true
       TEAM:
         restrictionType: AVAILABILITY
         enabled: false
       COMMUNITY:
         restrictionType: AVAILABILITY
-        enabled: false
+        enabled: true
```
Comment (sahilhindwani-harness) [inline 925-enforcement-service/src/main/resources/features/pipeline-features.yml:66]: I think we don't have template-service in our community bundle, could you please check once. @utkarshchoubey-harness  might be able to help in confirming this.
https://github.com/harness/harness-cd-community/blob/main/helm/harness/Chart.yaml

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]

**Comment by meenaravichandran1** [comment]
trigger uts

**Comment by meenaravichandran1** [comment]
trigger checkstylecodeformat

**Review by sahilhindwani-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56238)* status: 1 file(s) to review.

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56238)* status: 1 file(s) to review.

**Comment by meenaravichandran1** [comment]
trigger uts

**Comment by meenaravichandran1** [comment]
trigger ut7

**Comment by meenaravichandran1** [comment]
trigger ut0

**Comment by meenaravichandran1** [comment]
trigger ut0

**Comment by meenaravichandran1** [comment]
trigger ut0

**Comment by meenaravichandran1** [comment]
trigger ut0

**Pull request merged**